torch/distributed/fsdp/flat_param.py

```diff
@@ -1293,6 +1293,14 @@ def _get_padded_unsharded_flat_param(self) -> torch.Tensor:
                 unsharded_flat_param.dtype != self._fwd_bwd_param_dtype,
                 f"Expects full precision but got {self._fwd_bwd_param_dtype}",
             )
+            # For no-reshard-after-forward strategies, `_full_param_padded` may
+            # still be allocated from a previous forward. As we are forcing
+            # full precision here, the full-precision unsharded copy may be
+            # modified, invalidating the existing low-precision unsharded copy,
+            # so we should free it here to ensure a new all-gather for the next
+            # forward/backward computation to persist the modifications.
+            if flat_param._full_param_padded.untyped_storage().size() > 0:
```

Comment:
Won't this regress the scenario where people use `summon_full_params(writeback=False)`?